core/src/main/java/com/linecorp/armeria/client/retry/BackoffSpec.java

```diff
@@ -0,0 +1,300 @@
+ * {@link ExponentialBackoff}, {@link FixedBackoff} and {@link RandomBackoff}. Only one of them can be
+ * specified in the {@code specification}, otherwise {@link IllegalArgumentException} will occur.
+ * <ul>
+ *     <li>{@code exponential=[initialDelayMillis:maxDelayMillis:multiplier]}
+ *     is for {@link Backoff#exponential(long, long, double)}
+ *     (maxDelayMillis and multiplier can be omitted)</li>
+ *     <li>{@code fixed=[delayMillis]} is for {@link Backoff#fixed(long)}</li>
+ *     <li>{@code random=[minDelayMillis:maxDelayMillis]} is for {@link Backoff#random(long, long)}</li>
+ *     <li>{@code jitter=[minJitterRate:maxJitterRate]} is for {@link Backoff#withJitter(double, double)}
+ *     (if only one jitter value is specified, it will be used for {@link Backoff#withJitter(double)}</li>
+ *     <li>{@code maxAttempts=[maxAttempts]} is for {@link Backoff#withMaxAttempts(int)}</li>
+ * </ul>
+ * So for example, you can specify it such as {@code random=100:10000,jitter=-0.4:0.3,maxAttempts=100}.
+ * The order of options does not matter, and the {@code specification} needs at least an option. If you don't
+ * specify them, the values are set by default values. The default values are
+ * {@code exponential=200:10000:2.0,jitter=0.2,maxAttempts=10}. After the values are set by
+ * {@link #parse(String)}, you can create the {@link Backoff} by calling {@link #build()}.
+ */
+final class BackoffSpec {
+    private static final Splitter OPTION_SPLITTER = Splitter.on(',').trimResults();
+    private static final Splitter KEY_VALUE_SPLITTER = Splitter.on('=').trimResults();
+    private static final Splitter VALUE_SPLITTER = Splitter.on(':').trimResults();
+
+    private static final String[] ORDINAL = { "first", "second", "third" };
+
+    private static final long DEFAULT_INITIAL_DELAY_MILLIS = 200;
+    private static final long DEFAULT_MAX_DELAY_MILLIS = 10000;
+    private static final double DEFAULT_MULTIPLIER = 2.0;
+    private static final double DEFAULT_MIN_JITTER_RATE = -0.2;
+    private static final double DEFAULT_MAX_JITTER_RATE = 0.2;
+    private static final int DEFAULT_MAX_ATTEMPTS = 10;
+
+    private final String specification;
+
+    private boolean exponentialConfigured; // base option group.
+    private boolean fixedConfigured;       // base option group.
+    private boolean randomConfigured;      // base option group. only one of base options can be configured.
+    private boolean jitterConfigured;
+    private boolean maxAttemptsConfigured;
+
+    long initialDelayMillis = DEFAULT_INITIAL_DELAY_MILLIS;
+    long maxDelayMillis = DEFAULT_MAX_DELAY_MILLIS;
+    double multiplier = DEFAULT_MULTIPLIER;
+    double minJitterRate = DEFAULT_MIN_JITTER_RATE;
+    double maxJitterRate = DEFAULT_MAX_JITTER_RATE;
+    int maxAttempts = DEFAULT_MAX_ATTEMPTS;
+    long fixedDelayMillis;
+    long randomMinDelayMillis;
+    long randomMaxDelayMillis;
```

Comment:
If these fields are package-private for testing, please add `@VisibleForTesting` to them.